File: SourceCode/Debug Helpers/open.fda.gov/ApiDalc/DataObjects/DrugData.vb
#Region " Imports "

Imports ApiDalc.Enumerations
Imports Newtonsoft.Json.Linq ' OpenSource

#End Region

Namespace DataObjects

    ''' <summary>
    ''' Drug Data Object
    ''' </summary>
    ''' <remarks></remarks>
    Public Class DrugData

#Region " Public Properties "

        Public Property ActionDrug As ActionDrug

        Public Property DrugCumulativeDosageNumb As String

        Public Property DrugCumulativeDosageUnit As DrugCumulativeDosageUnit

        Public Property DrugTreatmentDuration As String

        Public Property DrugTreatmentDurationUnit As DrugTreatmentDurationUnit

        Public Property DrugIntervalDosageUnitNumb As String

        Public Property DrugIntervalDosageDefinition As DrugIntervalDosageDefinition

        Public Property DrugRecurreAdministration As DrugRecurreAdministration

        Public Property DrugAdditional As String

        Public Property DrugAuthorizationNumb As String

        Public Property DrugDosageForm As String

        Public Property DrugCharacterization As DrugCharacterization

        Public Property Drugadministrationroute As String

        Public Property DrugDosageText As String

        Public Property DrugStartDate As String

        Public Property DrugEndDate As String

        Public Property DrugStartDateFormat As String

        Public Property DrugEndDateFormat As String

        'Public Property drugcharacterization As String

        'Public Property medicinalproduct As String

        Public Property DrugIndication As String

        Public Property MedicinalProduct As String

        Public Property OpenFDA As OpenFdaData

#End Region

        ''' <summary>
        ''' Convert JSON Data
        ''' </summary>
        ''' <param name="jToken">Token</param>
        ''' <returns>List of Drug Data</returns>
        ''' <remarks></remarks>
        Public Shared Function ConvertJsonData(jToken As JToken) As List(Of DrugData)

            Dim data As New List(Of DrugData)

            If IsJTokenValid(jToken) Then

                For Each drug In jToken

                    Dim obj As New DrugData

                    Dim actionDrug As Integer = CInt(obj.ActionDrug)
                    Dim drugRecurreAdministraton As Integer = CInt(obj.DrugRecurreAdministration)
                    Dim drugIntervalDosageDefinition As Integer = CInt(obj.DrugIntervalDosageDefinition)
                    Dim drugTreatmentDurationUnit As Integer = CInt(obj.DrugTreatmentDurationUnit)
                    Dim drugCumulativeDosageUnit As Integer = CInt(obj.DrugCumulativeDosageUnit)
                    Dim drugCharacterization As Integer = CInt(obj.DrugCharacterization)

                    Integer.TryParse(CStr(drug("actiondrug")), actionDrug)
                    obj.ActionDrug = CType(actionDrug, Enumerations.ActionDrug)

                    Integer.TryParse(CStr(drug("drugrecurreadministration")), drugRecurreAdministraton)
                    obj.DrugRecurreAdministration = CType(drugRecurreAdministraton, Enumerations.DrugRecurreAdministration)

                    obj.DrugIntervalDosageUnitNumb = CStr(drug("drugintervaldosageunitnumb"))

                    Integer.TryParse(CStr(drug("drugintervaldosagedefinition")), drugIntervalDosageDefinition)
                    obj.DrugIntervalDosageDefinition = CType(drugIntervalDosageDefinition, Enumerations.DrugIntervalDosageDefinition)

                    obj.DrugTreatmentDuration = CStr(drug("drugtreatmentduration"))

                    Integer.TryParse(CStr(drug("drugtreatmentdurationunit")), drugTreatmentDurationUnit)
                    obj.DrugTreatmentDurationUnit = CType(drugTreatmentDurationUnit, Enumerations.DrugTreatmentDurationUnit)

                    obj.DrugCumulativeDosageNumb = CStr(drug("drugcumulativedosagenumb"))

                    Integer.TryParse(CStr(drug("drugcumulativedosageunit")), drugCumulativeDosageUnit)
                    obj.DrugCumulativeDosageUnit = CType(drugCumulativeDosageUnit, Enumerations.DrugCumulativeDosageUnit)

                    Integer.TryParse(CStr(drug("drugcharacterization")), drugCharacterization)
                    obj.DrugCharacterization = CType(drugCharacterization, Enumerations.DrugCharacterization)

                    obj.DrugAdditional = CStr(drug("drugadditional"))

                    obj.DrugAuthorizationNumb = CStr(drug("drugauthorizationnumb"))

                    obj.DrugIndication = CStr(drug("drugindication"))
                    obj.MedicinalProduct = CStr(drug("medicinalproduct"))
                    obj.Drugadministrationroute = CStr(drug("drugadministrationroute"))
                    obj.DrugDosageText = CStr(drug("drugdosagetext"))

                    obj.DrugStartDate = CStr(drug("drugstartdate"))
                    obj.DrugStartDateFormat = CStr(drug("drugstartdateformat"))
                    obj.DrugEndDate = CStr(drug("drugenddate"))
                    obj.DrugEndDateFormat = CStr(drug("drugenddateformat"))

                    obj.DrugDosageForm = CStr(drug("drugdosageform"))


                    'obj.drugcharacterization = drug("drugcharacterization")
                    'obj.medicinalproduct = drug("medicinalproduct")

                    obj.OpenFDA = OpenFdaData.ConvertJsonData(drug("openfda"))

                    data.Add(obj)

                Next

            End If

            Return data

        End Function

    End Class

End Namespace

File: SourceCode/AspCoreVersion/src/ShopAware.Core/DataObjects/DrugData.cs
using System.Collections.Generic;
using Newtonsoft.Json.Linq;
using ShopAware.Core.Enumerations;

namespace ShopAware.Core
{
    namespace DataObjects
    {
        /// <summary>
        ///     Drug Data Object
        /// </summary>
        /// <remarks></remarks>
        public class DrugData
        {
            #region Properties

            public ActionDrug ActionDrug { get; set; }

            public string DrugCumulativeDosageNumb { get; set; }

            public DrugCumulativeDosageUnit DrugCumulativeDosageUnit { get; set; }

            public string DrugTreatmentDuration { get; set; }

            public DrugTreatmentDurationUnit DrugTreatmentDurationUnit { get; set; }

            public string DrugIntervalDosageUnitNumb { get; set; }

            public DrugIntervalDosageDefinition DrugIntervalDosageDefinition { get; set; }

            public DrugRecurreAdministration DrugRecurreAdministration { get; set; }

            public string DrugAdditional { get; set; }

            public string DrugAuthorizationNumb { get; set; }

            public string DrugDosageForm { get; set; }

            public DrugCharacterization DrugCharacterization { get; set; }

            public string Drugadministrationroute { get; set; }

            public string DrugDosageText { get; set; }

            public string DrugStartDate { get; set; }

            public string DrugEndDate { get; set; }

            public string DrugStartDateFormat { get; set; }

            public string DrugEndDateFormat { get; set; }

            //Public Property drugcharacterization As String

            //Public Property medicinalproduct As String

            public string DrugIndication { get; set; }

            public string MedicinalProduct { get; set; }

            public OpenFdaData OpenFDA { get; set; }

            #endregion

            #region Public Methods

            /// <summary>
            ///     Convert JSON Data
            /// </summary>
            /// <param name="jToken">Token</param>
            /// <returns>List of Drug Data</returns>
            /// <remarks></remarks>
            public static List<DrugData> ConvertJsonData(JToken jToken)
            {
                var data = new List<DrugData>();

                if (Utilities.IsJTokenValid(jToken))
                {
                    foreach (var drug in jToken)
                    {
                        var obj = new DrugData();

                        var actionDrug = (int) obj.ActionDrug;
                        var drugRecurreAdministraton = (int) obj.DrugRecurreAdministration;
                        var drugIntervalDosageDefinition = (int) obj.DrugIntervalDosageDefinition;
                        var drugTreatmentDurationUnit = (int) obj.DrugTreatmentDurationUnit;
                        var drugCumulativeDosageUnit = (int) obj.DrugCumulativeDosageUnit;
                        var drugCharacterization = (int) obj.DrugCharacterization;

                        int.TryParse((Utilities.GetJTokenString(drug, "actiondrug")), out actionDrug);
                        obj.ActionDrug = (ActionDrug) actionDrug;

                        int.TryParse((Utilities.GetJTokenString(drug, "drugrecurreadministration")), out drugRecurreAdministraton);
                        obj.DrugRecurreAdministration = (DrugRecurreAdministration) drugRecurreAdministraton;

                        obj.DrugIntervalDosageUnitNumb = (Utilities.GetJTokenString(drug, "drugintervaldosageunitnumb"));

                        int.TryParse((Utilities.GetJTokenString(drug, "drugintervaldosagedefinition")), out drugIntervalDosageDefinition);
                        obj.DrugIntervalDosageDefinition = (DrugIntervalDosageDefinition) drugIntervalDosageDefinition;

                        obj.DrugTreatmentDuration = (Utilities.GetJTokenString(drug, "drugtreatmentduration"));

                        int.TryParse((Utilities.GetJTokenString(drug, "drugtreatmentdurationunit")), out drugTreatmentDurationUnit);
                        obj.DrugTreatmentDurationUnit = (DrugTreatmentDurationUnit) drugTreatmentDurationUnit;

                        obj.DrugCumulativeDosageNumb = (Utilities.GetJTokenString(drug, "drugcumulativedosagenumb"));

                        int.TryParse((Utilities.GetJTokenString(drug, "drugcumulativedosageunit")), out drugCumulativeDosageUnit);
                        obj.DrugCumulativeDosageUnit = (DrugCumulativeDosageUnit) drugCumulativeDosageUnit;

                        int.TryParse((Utilities.GetJTokenString(drug, "drugcharacterization")), out drugCharacterization);
                        obj.DrugCharacterization = (DrugCharacterization) drugCharacterization;

                        obj.DrugAdditional = (Utilities.GetJTokenString(drug, "drugadditional"));

                        obj.DrugAuthorizationNumb = (Utilities.GetJTokenString(drug, "drugauthorizationnumb"));

                        obj.DrugIndication = (Utilities.GetJTokenString(drug, "drugindication"));
                        obj.MedicinalProduct = (Utilities.GetJTokenString(drug, "medicinalproduct"));
                        obj.Drugadministrationroute = (Utilities.GetJTokenString(drug, "drugadministrationroute"));
                        obj.DrugDosageText = (Utilities.GetJTokenString(drug, "drugdosagetext"));

                        obj.DrugStartDate = (Utilities.GetJTokenString(drug, "drugstartdate"));
                        obj.DrugStartDateFormat = (Utilities.GetJTokenString(drug, "drugstartdateformat"));
                        obj.DrugEndDate = (Utilities.GetJTokenString(drug, "drugenddate"));
                        obj.DrugEndDateFormat = (Utilities.GetJTokenString(drug, "drugenddateformat"));

                        obj.DrugDosageForm = (Utilities.GetJTokenString(drug, "drugdosageform"));

                        obj.OpenFDA = OpenFdaData.ConvertJsonData((JObject) Utilities.GetJTokenObject(drug, "openfda"));

                        data.Add(obj);
                    }
                }

                return data;
            }

            #endregion
        }
    }
}